### PR TITLE
[Merged by Bors] - chore: split `MetricSpace/Baire`

### PR DESCRIPTION
**Pull request by urkud**
* move definition to `Topology/Defs/Basic`;
* move lemmas to `Topology/Baire/Lemmas`;
* move instances to `Topology/Baire/CompleteMetrizable` and `Topology/Baire/LocallyCompactRegular`;
* assume `[UniformSpace X] [IsCountablyGenerated (𝓤 X)]` instead of `[PseudoMetricSpace X]` in the 1st theorem.

This way `Lemmas` file does not depend on analysis.

---
I'm going to fix imports with the help of CI.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Baire/CompleteMetrizable.lean [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import Mathlib.Topology.Metrizable.Uniformity
+
+#align_import topology.metric_space.baire from "leanprover-community/mathlib"@"b9e46fe101fc897fb2e7edaf0bf1f09ea49eb81a"
+
+/-!
+# First Baire theorem
+
+In this file we prove that a completely metrizable topological space is a Baire space.
+Since `Mathlib` does not have the notion of a completely metrizable topological space yet,
+we state it for a complete unifiorm space with countably generated uniformity filter.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Baire/CompleteMetrizable.lean:16]:
```suggestion
we state it for a complete uniform space with countably generated uniformity filter.
```

File: Mathlib/Topology/Baire/CompleteMetrizable.lean [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+open Filter EMetric Set
+open scoped Topology Uniformity ENNReal
+
+variable {X : Type*} [UniformSpace X] [CompleteSpace X] [(𝓤 X).IsCountablyGenerated]
+
+/-- **First Baire theorem**: a completely metrizable topological space has Baire property.
+
+Since `Mathlib` does not have the notion of a completely metrizable topological space yet,
+we state it for a complete unifiorm space with countably generated uniformity filter. -/
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Baire/CompleteMetrizable.lean:27]:
```suggestion
we state it for a complete uniform space with countably generated uniformity filter. -/
```

File: Mathlib/Topology/Baire/Lemmas.lean [hunk 3]
```diff
@@ -3,22 +3,25 @@ Copyright (c) 2019 Sébastien Gouëzel. All rights reserved.
...
 /-!
-# Baire theorem
+# Baire spaces
 
-In a complete metric space, a countable intersection of dense open subsets is dense.
+A topological space is called a *Baire space*
+if a countable intersection of dense open subsets is dense.
+Baire theorems say that all completely metrizable spaces
+and all locally compact regular spaces are Baire spaces.
+We prove the theorems in other files.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Baire/Lemmas.lean:17]: Please add which files

File: Mathlib/Topology/Defs/Basic.lean [hunk 4]
```diff
@@ -166,3 +166,11 @@ scoped[Topology] notation (name := closure_of) "closure[" t "]" => @closure _ t
 /-- Notation for `Continuous` with respect to a non-standard topologies. -/
 scoped[Topology] notation (name := Continuous_of) "Continuous[" t₁ ", " t₂ "]" =>
   @Continuous _ _ t₁ t₂
+
+/-- The property `BaireSpace α` means that the topological space `α` has the Baire property:
+any countable intersection of open dense subsets is dense.
+Formulated here when the source space is ℕ (and subsumed below by `dense_iInter_of_isOpen` working
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Defs/Basic.lean:172]: Not "below" anymore

**Review by Ruben-VandeVelde — APPROVED** [review]
This seems reasonable

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**